Shared/Supporting Files/Views/CategoryView.swift

```diff
@@ -24,20 +24,47 @@ struct CategoryView: View {
     /// The search query in the search bar.
     @Binding private(set) var query: String
     
+    /// The samples to display in the name section of the search list.
+    @State private var nameSearchResults: [Sample] = []
+    
+    /// The samples to display in the description section of the search list.
+    @State private var descriptionSearchResults: [Sample] = []
+    
+    /// The samples to display in the tags section of the search list.
+    @State private var tagsSearchResults: [Sample] = []
```

Comment:
Since these are all related, you might consider combining them:

```swift
struct SearchResult {
    var nameMatches: [Sample]
    var descriptionMatches: [Sample]
    var tagMatches: [Sample]
}

@State private var searchResult: SearchResult?
```